concepts/regular-expressions/about.md

```diff
@@ -0,0 +1,160 @@
+
+```javascript
+const funnyQuote =
+  'If you see someone crying, ask if it is because of their haircut.';
+const regex1 = /someone/;
+const regex2 = /happy/;
+
+funnyQuote.match(regex1); // ["someone", index: 3, input: "If you see someone crying, ask if it is because of their haircut.", groups: undefined]
+
+funnyQuote.match(regex2); // null
+```
+
+When the Global Search flag `/g` is present, instead of getting the only match alongside useful information such as the index or input, the method returns all of the occurances displayed in the array:
+
+```javascript
+const funnyQuote =
+  'If you see someone crying, ask if it is because of their haircut.';
+
+const regex3 = /if/gi;
+
+funnyQuote.match(regex3); // ["If", "if"];
+```
+
+### Replace
+
+The [replace()][mdn-regex-replace] method in JavaScript allows us to search for a value within a given string, and replacing it with a desired new value.
+
+```javascript
+string.replace(searchValue, newValue);
+```
+
+The pattern for searching the substitution can be a single string, or a regular expression.
+
+```javascript
+let string = 'I like dogs!';
+let result = string.replace('dogs', 'cats');
+
+let string = 'I would love to travel to Japan';
+let result = string.replace(/Japan/g, 'Hawaii');
+```
+
+Moreover, we can apply a function on the replacement position in order to make additional changes to each value.
+
+```javascript
+let text = 'Say hello to the chatbot.';
+let result = text.replace(/chatbot|hello/gi, function (word) {
+  return word.toUpperCase();
+});
+// -> "Say HELLO to the CHATBOT"
+```
+
+### Split
+
+The [split()][mdn-regex-split] method in JavaScript represents a different way of using and manipulating strings with regular expressions.
+
+```javascript
+const str = 'hello,user.how are.you';
+
+const result = str.split(/[,.\s]/);
+
+console.log(result);
+//--> ['hello', 'user', 'how', 'are', 'you']
+```
+
+## 5. Performance with Regex
+
+Regarding performance, both of them create a RegExp object. The main difference is how often the regex is compiled:
+
+- With `Regular Expression Literal`: one time during initial code parsing and compiling
+- With `RegExp()` syntax: Every time new Object gets created.
+
+According to [this](https://stackoverflow.com/a/32523333)answer from Alexander Abakumov in StackOverflow, using literal syntax may be a better option not only because of performance, but also for simplicity and readability.
+
+1. It is shorter and doesn’t force you to think in terms of class-like constructors.
+2. When using the `RegExp()` constructor, you also need to escape quotes and double-escape backslashes. It makes regular expressions that are hard to read and understand by their nature even more harder.
+
+Nevertheless, for this particular challenge, we are practising with both methods. Hence you are able to decide which one to apply depending on the occasion.
+
+## 6. Useful Resources
```

Comment:
This headline can be removed. The list below will not be rendered. Instead, the website will render the content of the links.json file with a "Read more" headline.